### ISSUE TITLE
Google's networking inside datacenter and to the outside

### ISSUE THREAD

**Issue by wangyinfeng** [wangyinfeng/notes#4]
https://www.nextplatform.com/2017/07/17/google-wants-rewire-internet/

Google carries about 25 percent of Internet traffic today. In other words, one out of four bytes that are delivered to end users across the Internet originate from Google. 

B4 is our private datacenter network interconnect, and B2 is what connects our datacenter to the public Internet.

(The traditional routing) is focused on individual boxes, and essentially what this means is that they have to take a very **local view** about connectivity. As soon as they can find a path between a source and a destination, two computers that want to talk to each other across the network, then BGP is happy. It doesn’t try to find the best path, and it doesn’t try to do dynamic optimization. 
basically what Espresso does is it actually pulls the routing intelligence outside of individual routers into a server pool, where we can do offline analysis of the data.

Espresso is serving 20 percent of the traffic

**Comment by wangyinfeng** [comment]
https://www.linkedin.com/in/vahdat/

https://research.google.com/pubs/AminVahdat.html

**Comment by wangyinfeng** [comment]
https://www.nextplatform.com/2015/06/19/inside-a-decade-of-google-homegrown-datacenter-networks/
